lib/esm/embed.rb

```diff
@@ -19,6 +19,79 @@ def self.build(type = nil, **, &)
       ESM::Embed.new(type, **, &)
     end
 
+    # This is defensively written because this can receive user provided data
+    def self.from_hash(hash, &block)
+      raise TypeError, "Expected Hash, got #{hash.class}" unless hash.is_a?(Hash)
+
+      hash = hash.with_indifferent_access
+      new do |embed|
+        ###########
+        # Author
+        # Supports string or HashMap with full options
+        if (author = hash[:author]) && author.present?
```

Comment:
Why is multiline unless statement bad but multiline if statement ok?

Reply:
I'll expand a bit more about what I mentioned in the other comment. Again, nothing is wrong or "bad". It is down to personal choice and how my brain reads these things. 

Given the following:
```rb
if number.positive?
  # Do something if number is positive
end
```

This has a very predictable flow: "If condition is true, do something 

Let's rewrite the same flow but using an `unless`
```rb
unless number.negative?
  # Do something if number is positive
end
```

This has a less explicit workflow. Reading it as "unless this thing is true, do this" is easy, but it has always required me to make sure I haven't confused myself in my head. 

I've found that when writing code, designing data structures, working with databases, and even talking with people that if you stay "positive", your life will be significantly easier. Whereas I can't say I'm successful staying positive in life, I can say that this has made working with older code so much easier. 

Ask yourself, which is easier to read? `complete: true`, or `incomplete: false`